src/chat/RoomContext.tsx

```diff
@@ -60,6 +74,12 @@ const RoomProvider = function ({
       room.current!.moveMarker();
     });
 
+    room.current!.on('marker_moved', (marker: Marker) => {
+      if (marker.userId === room.current!.getUser().id) {
```

Comment:
what is this syntax doing? i.e `room.current!.`

Reply:
It makes typescript stop complaining about `room.current` being `null` when it cannot possibly be `null` in this context. The alternative would be to add a big if statement, but that would be an even uglier solution compared to the exclamation mark.